src/xunit.console.netcore/Visitors/StandardOutputVisitor.cs

```diff
@@ -116,6 +145,22 @@ protected override bool Visit(ITestFinished testFinished)
                     Console.WriteLine("   {0} [FINISHED] Time: {1}s", XmlEscape(testFinished.Test.DisplayName), testFinished.ExecutionTime);
                 }
             }
+            if (!runningTests.TryRemove(testFinished.Test.DisplayName, out long elapsed))
+            {
+                lock (consoleLock)
```

Comment:
Console is internally thread safe for reads/writes I believe. You should not need any lock -- unless you want to write two consecutive lines, and even then you would need all of xunit to participate in the lock.